### DISCUSSION TITLE
Follow references inside Forge

### DISCUSSION THREAD

**Discussion by Thaodan** [magit/forge#521] [Ideas]
Quite often bug or PR's contain references to other items.

When following those they are opened in program assigned with the http-adress type.
I would be useful if the would be opened in a forge buffer instead or an option to do so.

The alternative would be to open the reference in the browser or go back and look for the reference.

**Comment by tarsius** [comment]
The bug-reference mode opens the topic in a browser. You could also press `n v t` to visit the topic using forge; or you could add a shorter binding for `forge-visit-topic`. Doing that properly, i.e., only when the cursor is on a topic number, is a bit more involved, which is why I haven't gotten around to that yet.

**Comment by ccqpein** [comment]
If I understand @Thaodan right, I have the same idea today. Like when I check the PR I want check the `change files` in my emacs rather than switch the context to browser. 

Further more I implement a draft version in my local. 

Firstly, in forge's `forge-insert-topic-refs` function, I need to add the value to the `magit-section` (in `magit-insert-section` macro). This can merge first because the `topic-refs` section name won't match the `magit-diff--dwim` anyway. [Draft PR](https://github.com/magit/forge/pull/522)

Secondly, change magit side `magit-diff--dwim`, just add one line in the ending of`magit-saction-case`, `(topic-refs (oref it value))`.  [Draft PR](https://github.com/magit/magit/pull/4770)

Then in forge topic buffer, on the line `Refs: abcde...fghij`, shortcut `d d` (`magit-diff-dwim`) will into the magit diff buffer.

**Reply by tarsius** [reply]
@ccqpein said:

Just updated the comment upper with two draft PRs. I think it implement this feature, but I am not very sure the breaking change (section (topic-refs) become (topic-refs . abcde...fghij)) will affect somewhere else those using the topic-refs section.

**Reply by tarsius** [reply]
I don't think this is what @Thaodan asked for.

Furthermore, what *you* want to do is already possible, except that you have to move to the `Commits` line, instead of the `Refs: ...` line before pressing `d d`. It might make sense to also support `d d` in that second place *or* even anywhere in that buffer. I'll think about that at a later time and respond in one of your pull-requests.

**Reply by ccqpein** [reply]
Sure, thanks @tarsius

**Comment by tarsius** [comment]
I've finally implemented this in 3662e218c808bd3497a8aa05f3ccb533fa60eaa4.
